platform/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	TypeSourced = "Sourced"
+	TypeApplied = "Applied"
+
+	ReasonSourceFailed     = "SourceFailed"
+	ReasonSourceSuccessful = "SourceSuccessful"
+	ReasonApplyFailed      = "ApplyFailed"
+	ReasonApplySuccessful  = "ApplySuccessful"
+)
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+	// PackageName contains the desired OLM-based Operator package name
+	// that is defined in an existing CatalogSource resource in the cluster.
+	//
+	// This configured package will be managed with the cluster's lifecycle.
+	// +kubebuilder:validation:Required
+	// +required
+	PackageName string `json:"packageName"`
+}
+
+// ActiveBundleDeployment references a BundleDeployment resource.
+type ActiveBundleDeployment struct {
+	// Name is the metadata.name of the referenced BundleDeployment object.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator
+type PlatformOperatorStatus struct {
+	// Conditions represent the latest available observations of a platform operator's current state.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// ActiveBundleDeployment is the reference to the BundleDeployment resource that's
+	// being managed by this PO resource.
+	// +optional
+	ActiveBundleDeployment ActiveBundleDeployment `json:"activeBundleDeployment,omitempty"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope="Cluster",shortName={"platform","platforms"}
```

Comment:
Do we know whether these short names won't conflict with anything already defined in openshift? Might be worth spinning up a cluster to double-check, but my best guess is these will need to be removed.

Reply:
I booted up a cluster in 4.12 and it didn't have anything with the `platform` or `platforms` short names. Maybe it is worth keeping?

Reply:
I'm hesitant about these shortname, see my other notes

Reply:
Yeah, let's just drop the short name imo.